model/circulationevent.py

```diff
@@ -89,7 +131,10 @@ class CirculationEvent(Base):
 
     @classmethod
     def log(cls, _db, license_pool, event_name, old_value, new_value,
-            start=None, end=None, foreign_patron_id=None):
+            start=None, end=None, library=None, location=None):
+        """Log a CirculationEvent to the database, assuming it
+        hasn't already been recorded.
+        """
```

Comment:
This method was missing a test so I added one to make sure there's no regression here (apart from the removal of foreign_patron_id).